crates/biome_js_analyze/tests/specs/nursery/useTanStackQueryExhaustiveDeps/import_detection.js

```diff
@@ -0,0 +1,77 @@
+// TanStack: Properly imported from @tanstack/react-query
+import { useQuery } from '@tanstack/react-query';
+
+function ValidComponent1({ userId }) {
+  const query = useQuery({
+    queryKey: ['users'],
+    queryFn: () => fetchUser(userId) // Should report missing userId
+  });
+}
+
+// TanStack: Namespace import
+import * as Query from '@tanstack/react-query';
+
+function ValidComponent2({ userId }) {
+  const query = Query.useQuery({
+    queryKey: ['users'],
+    queryFn: () => fetchUser(userId) // Should report missing userId
+  });
+}
+
+// TanStack: Experimental package
+import { useQuery } from '@tanstack/react-query/experimental';
+
+function ValidComponent3({ userId }) {
+  const query = useQuery({
+    queryKey: ['users'],
+    queryFn: () => fetchUser(userId) // Should report missing userId
+  });
+}
+
+// TanStack: Legacy package name
+import { useQuery } from 'react-query';
+
+function ValidComponent4({ userId }) {
+  const query = useQuery({
+    queryKey: ['users'],
+    queryFn: () => fetchUser(userId) // Should report missing userId
+  });
+}
```

Comment:
TanStack Query <= v4 had the different signature of `useQuery`. It would be great if the rule also support that cases.

https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5#supports-a-single-signature-one-object